### PR TITLE
add a warning for deletion of a  systemdatabase on a cluster.

### PR DESCRIPTION
**Pull request by Poltergeist**
Add a warning if the current database is an internal cluster database
and _stats is available on that note. _stats is only available if
it is a cluster internal node.
The warning is only shown for databases with the name:
  - _replicator
  - _users
  - dbs
  - nodes

### CODE REVIEW

**Comment by garrensmith** [comment]
@Poltergeist @robertkowalski this looks great. Could you add a few tests for this.

**Comment by Poltergeist** [comment]
@garrensmith I'll add tests asap in the next days.

**Comment by Poltergeist** [comment]
Added test cases for the parse function of the new model.

File: app/addons/documents/views.js [hunk 1]
```diff
@@ -145,9 +145,16 @@ function (app, FauxtonAPI, Components, Documents,
     template: "addons/documents/templates/delete_database_modal",
     initialize: function (options) {
       this.database = options.database;
+      this.isSystemDatabase = options.isSystemDatabase;
       FauxtonAPI.Events.on('database:delete', this.showDeleteDatabase, this);
     },
 
+    serialize: function () {
+      return {
+        isSystemDatabase: this.isSystemDatabase
```
Comment (garrensmith) [inline app/addons/documents/views.js]: This serialize function also needs to return database. You need to make the return this:

    return {
      isSystemDatabase: this.isSystemDatabase,
      database: this.database
    };

**Comment by garrensmith** [comment]
@Poltergeist there is a slight bug. When I try delete a database the database name doesn't appear in the modal. See screenshot. I've added a comment on how to fix it. Other than that it is looking good. Once the bug is fixed we can merge

**Comment by Poltergeist** [comment]
@garrensmith added the test and also merged the master branch into this branch.

**Comment by garrensmith** [comment]
Thanks for this @Poltergeist this is great. I've merged it in.

**Comment by kxepal** [comment]
cassim database is missed in the list.

**Comment by Poltergeist** [comment]
I'll add the cassim database. I just didn't know about the cassim database.

**Comment by kxepal** [comment]
No worries. Nobody known (: Thought it might be also good idea to make all system databases with leading `_` to make sure.

**Comment by Poltergeist** [comment]
I added the cassim database in another pull request. Test for the _ prefixed names could be a good idea.

**Comment by kxepal** [comment]
@Poltergeist [COUCHDB-2619](https://issues.apache.org/jira/browse/COUCHDB-2619) fixed so all system databases are starts with underscore no matter on which iface they are. Could you please submit new PR? I'd tried by my own, but I don't understand what is going on in `IsSystemDatabaseModel`, why there is need to request `_stats` (which I strongly believe causes [COUCHDB-2607](https://issues.apache.org/jira/browse/COUCHDB-2607) ) when you could just GET app.host and look if uuid is in the response and how to turn it into simple property of Database model without ruining everything. Thanks (:

**Comment by robertkowalski** [comment]
@Poltergeist @kxepal will take care of that now!

**Pull request closed without merging**